mypy/checker.py

```diff
@@ -1459,6 +1461,9 @@ def check_multi_assignment(self, lvalues: List[Lvalue],
         rvalue_type = self.expr_checker.accept(rvalue)  # TODO maybe elsewhere; redundant
         undefined_rvalue = False
 
+        if isinstance(rvalue_type, UnionType) and len(rvalue_type.relevant_items()) == 1:
+            rvalue_type = rvalue_type.relevant_items()[0]
```

Comment:
Yeah, that was something that I was thinking about as I was writing this.  I could change this to:
```py
if isinstance(rvalue_type, UnionType):
  relevant_items = rvalue_type.relevant_items()
  if len(relevant_items) == 1:
    rvalue_type = relevant_items[0]
```
but I'm not sure that's particularly nicer?  If it seems better to you, then I'm happy to do it.